src/rpdk/core/contract/resource_client.py

```diff
@@ -315,11 +314,26 @@ def is_property_in_path(key, paths):
                 return True
         return False
 
+    @staticmethod
+    def get_value_by_key_path(model, key_path):
+        if isinstance(key_path, (tuple, list)):
+            value = model
+            for path in key_path:
+                value = value[path]
+            return value
+        return model[key_path]
+
     def validate_update_example_keys(self, unique_identifiers, update_example):
         for primary_identifier in self.primary_identifier_paths:
             if primary_identifier in self.create_only_paths:
-                primary_key = fragment_list(primary_identifier, "properties")[0]
-                assert update_example[primary_key] == unique_identifiers[primary_key], (
+                primary_key_path = fragment_list(primary_identifier, "properties")
+                update_example_pk_value = self.get_value_by_key_path(
+                    update_example, primary_key_path
+                )
+                unique_identifiers_pk_value = self.get_value_by_key_path(
+                    unique_identifiers, primary_key_path
+                )
+                assert update_example_pk_value == unique_identifiers_pk_value, (
```

Comment:
I assume that `get_value_by_key_path -> Dict` if so then will there be an ordering problem of the keys in the map?

Reply:
`get_value_by_key_path` returns the root-node value of that key path, so it should be a number or string. 

The original implementation used `key_path[0]` to grab the value, which will definitely return a Dict when the key is nested.